sync/upstream.py

```diff
@@ -721,13 +721,22 @@ def status_changed(git_gecko, git_wpt, sync, context, status, url, sha):
     if status == "pending":
         # Never change anything for pending
         return landed
-
-    if status == "success":
+    previous_status = sync.last_pr_check
+    success, non_success = env.gh_wpt.status_checks_pass(
+        sync.pr,
+        exclude=["upstream/gecko"]
+    )
+    if success:
+        sync.last_pr_check = "success"
+    else:
+        states = {item.state for item in non_success}
+        sync.last_pr_check = "failure" if "failure" in states else "pending"
+    if success and previous_status != "success":
```

Comment:
I think we should try the landing irrespective of the previous status, but only  add a comment if something changed.